lib/ricohapi/mstorage/client.rb

```diff
@@ -90,7 +92,18 @@ def delete(media_id)
 
       # PUT /media/{id}/meta/user/{key}
       def add_meta(media_id, user_meta)
-        # TODO: do something
+        if user_meta.empty?
+          raise Error.new("user_meta is empty: nothing to request.")
+        end
+        user_meta.each do |k, v|
+          if k.empty? || !valid?(v)
+            raise Error.new("Invalid parameter: {#{k} => #{v}}")
+          end
```

Comment:
That looks much better. I'll use that code.
Also, I'm gonna change the method name.